xapian-applications/omega/query.cc

```diff
@@ -1228,6 +1233,10 @@ get_subdbs()
     return subdbs;
 }
 
+// mersenne twister for RNG
+random_device rd;
+mt19937 rng(rd());
```

Comment:
It's cleaner to make file-scope objects such as these `static` (unless they actually need to be visible to other source files).

What this code does (as you're probably aware) is to generate a random number using `std::random_device` and use that to seed the MT19037 pseudo-random number generator.  This will happen early in program initialisation, before `main()` is called.

In most cases, `std::random_device` will use the system's hardware entropy source, so doing this on every run of Omega will use up a few bytes of entropy for each run (even when using templates which never use `$random` or always call `$srandom` before calling `$random`).  Servers can sometimes struggle to generate enough entropy, so consuming it for no reason is potentially anti-social and can cause other programs which need entropy to block waiting for more to be harvested.

So I think we don't want to be seeding from `std::random_device` here.  Either we track whether `$srandom` has been called and if not seed from `std::random_device` the first time `$random` is called, or just allow `$srandom` with no argument to mean "seed from `std::random_device`" and then leave it up to the template author to decide if and how to seed the the PRNG.

Reply:
Well, I clearly didn't think that far. I implemented the first option that you suggested. Thanks for the input!